### DISCUSSION TITLE
Make a resource property being able to refer to another property of that (or parent) resource.

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#4357] [Engine Core]
For example I have a Theme (gui).
I want to set some of its properties to one value.
Say for FileDialog I want file_icon_modulate and folder_icon_modulate refer to property color1 that I created in the theme script. 
That means if I change color1, when both file_icon_modulate and folder_icon_modulate are  changed automatically.
This would be useful.
I don't want to use a script for creating a setter or so.
This should be done completely in the editor (Inspector), except for creating properties.

Any proposals or issues-requests (including closed ones) I can read?

**Comment by Mickeon** [comment]
So, some sort of inheritance for the individual Resources themselves? Kind of similarly to how Scene inheritance works.

**Reply by me2beats** [reply]
Kinda references to variables I guess, not inheritance

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot-proposals/issues/3257.

Note that Resource inheritance has been considered (or adding a Color resource type), but neither are likely to be implemented for 4.0.
